base/shell/explorer/traywnd.cpp

```diff
@@ -813,20 +853,35 @@ class CTrayWindow :
 
         case ID_SHELL_CMD_TILE_WND_H:
             appbar_notify_all(NULL, ABN_WINDOWARRANGE, NULL, TRUE);
+            if (g_Arrangement == NONE)
+            {
+                BackupWindowPos();
+            }
             TileWindows(NULL, MDITILE_HORIZONTAL, NULL, 0, NULL);
             appbar_notify_all(NULL, ABN_WINDOWARRANGE, NULL, FALSE);
+            g_Arrangement = TILED;
```

Comment:
... while `BackupWindowPos()` _might_ take a parameter specifying the arrangement value `TILED` or `CASCADED`, and set `g_Arrangement` accordingly.

Reply:
I'm not sure what you mean here.
Do you mean to move the setting of g_Arrangement to BackupWindowPos()?
Wouldn't that make the logic quite different?
According to the current code, g_Arrangement is to be set regardless of whether a backup was made (g_Arrangement == NONE).
Moving it to BackupWindowPos() would make it only change the arrangement variable if a backup *was* made, no?

Reply:
yes indeed you're correct. Forget my suggestion here.